plugin/trino-mongodb/src/test/java/io/trino/plugin/mongodb/BaseMongoConnectorTest.java

```diff
@@ -584,6 +584,71 @@ public void testAddColumnConcurrently()
         throw new SkipException("TODO");
     }
 
+    @Test
+    public void testNativeQuerySimple()
+    {
+        assertThat(query("SELECT * FROM TABLE(mongodb.system.query(database => 'tpch', collection => 'region', filter => '{ regionkey: 1 }'))"))
+                .matches("SELECT * FROM region WHERE regionkey = 1");
+
+        // Empty filter '{}' returns all rows
+        assertThat(query("SELECT * FROM TABLE(mongodb.system.query(database => 'tpch', collection => 'region', filter => '{}'))"))
+                .matches("SELECT * FROM region");
+    }
+
+    @Test
+    public void testNativeQueryFilterAndWhere()
```

Comment:
nit: Can we have query with filter applied on same column ? The PTF would have a larger subset and query filter will select a subset and also one where the filter would be a disjunct  ?

Reply:
Good catch. I introduced `andPredicate` method to bind two conditions (PTF and WHERE) with `$and` operator.